grcs/0020-knowledge-graph.md

```diff
@@ -32,164 +32,176 @@ Data is encoded using Protobufs. Protocol Buffers are a language-neutral, platfo
+| ID            | ID   | The ID for this relation                                                      |
+| Type          | ID   | The ID of the relation type                                                   |
+| From entity   | ID   | The entity ID this relation is from                                             |
+| From property | ID   | An optional property ID for creating a relation from a specific property value  |
+| To entity     | ID   | The entity ID this relation is pointing to                                      |
+| To space      | ID   | An optional space ID to link to an entity specifically in a given space         |
+| Entity        | ID   | A new entity ID that can be used to add properties and types to the relation    |
+| Index         | Text | An alphanumeric fractional index describing the relative position of this item |
 
-Examples:
-* "MMMM do, yyyy" - July 4th, 2024
-* "M/d/yyyy" - 7/4/2024
-* "h:mma" - 4:45pm
-* "EEEE h:mma" - Friday 5:00pm
 
-#### 5.6 Point
+ID, Type, From entity, To entity, and Entity are required. Linked entities don't have to be created before they're referenced by relations. A globally unique ID must be created for the Entity though no additional information has to be provided for the relation entity. This way there is already an ID available for adding data about the relation.
 
-A point is a location specified with x and y coordinates string encoded. Points can be used for geo locations or other cartesian coordinate systems. Clients may choose how many dimensions to support. Example: `"12.554564, 11.323474"`
+The lightweight relation objects are immutable. They can only be created and deleted. The relation entity can be updated to add/remove properties or add/remove types.
+
+#### 8.1 Relation Types
+
+A Relation Type is both a property and a type, but specifically for defining one to one, one to many, and many to many relations. A Relation Type can have a name like `Team member`, `City` or a more graph style name like `Born in`. A relation must have exactly one Relation Type.
+
+| Name          | Type      | ID                     |
+| ------------- |-----------|----------------------- |
+| Relation Type | Type      | 0509279a-713d-4667-a412-24fe3757f554 |
+
+A Relation Type is also of type `Type` and has the following attributes:
+
+| Name                     | Type              | Description                                                                       |
+| ------------------------ |-------------------| --------------------------------------------------------------------------------- |
+| Relation value types     | Relation \<Type>  | A list of types that the relation To entity may have. Used as a hint              |
+| One max                  | Checkbox          | Whether to limit the number of relations from a given entity to no more than 1    |
+| Cascades                 | Checkbox          | Whether to move and delete the to entity and relation entity with the from entity |
```

Comment:
@yanivtal do we need a distinction between `move` and `delete` for cascade? Are there valid use-cases?